crates/core_arch/src/arm/neon/mod.rs

```diff
@@ -1136,12 +1156,595 @@ pub unsafe fn vtbx4_p8(a: poly8x8_t, b: poly8x8x4_t, c: uint8x8_t) -> poly8x8_t
+}
+
+/// Move vector element to general-purpose register
+#[inline]
+#[target_feature(enable = "neon")]
+#[cfg_attr(target_arch = "arm", target_feature(enable = "v7"))]
+#[rustc_args_required_const(1)]
+// see: https://clang.godbolt.org/z/J5xS2T
+//#[cfg_attr(test, assert_instr(umov, imm5 = 0))]
+#[cfg_attr(test, assert_instr(fmov, imm5 = 0))]
+pub unsafe fn vgetq_lane_u32(v: uint32x4_t, imm5: i32) -> u32 {
+    if (imm5) < 0 || (imm5) > 3 {
+        unreachable_unchecked()
+    }
+    let imm5 = (imm5 & 0b11) as u32;
+    simd_extract(v, imm5)
+}
+
+/// Move vector element to general-purpose register
+#[inline]
+#[target_feature(enable = "neon")]
+#[cfg_attr(target_arch = "arm", target_feature(enable = "v7"))]
+#[rustc_args_required_const(1)]
+#[cfg_attr(test, assert_instr(umov, imm5 = 0))]
+pub unsafe fn vget_lane_u8(v: uint8x8_t, imm5: i32) -> u8 {
+    if (imm5) < 0 || (imm5) > 7 {
+        unreachable_unchecked()
+    }
+    let imm5 = (imm5 & 7) as u32;
+    simd_extract(v, imm5)
+}
+
+/// Duplicate vector element to vector or scalar
+// int8x16_t vdupq_n_s8 (int8_t value)
+#[inline]
+#[target_feature(enable = "neon")]
+#[cfg_attr(target_arch = "arm", target_feature(enable = "v7"))]
+#[cfg_attr(all(test, target_arch = "arm"), assert_instr(dup))]
+#[cfg_attr(all(test, target_arch = "aarch64"), assert_instr(dup))]
+pub unsafe fn vdupq_n_s8(value: i8) -> int8x16_t {
+    int8x16_t(
+        value, value, value, value, value, value, value, value, value, value, value, value, value,
+        value, value, value,
+    )
+}
+
+/// Duplicate vector element to vector or scalar
+// uint8x16_t vdupq_n_u8 (uint8_t value)
+#[inline]
+#[target_feature(enable = "neon")]
+#[cfg_attr(target_arch = "arm", target_feature(enable = "v7"))]
+#[cfg_attr(all(test, target_arch = "arm"), assert_instr(dup))]
+#[cfg_attr(all(test, target_arch = "aarch64"), assert_instr(dup))]
+pub unsafe fn vdupq_n_u8(value: u8) -> uint8x16_t {
+    uint8x16_t(
+        value, value, value, value, value, value, value, value, value, value, value, value, value,
+        value, value, value,
+    )
+}
+
+/// Duplicate vector element to vector or scalar
+// uint8x16_t vmovq_n_u8 (uint8_t value)
+#[inline]
+#[target_feature(enable = "neon")]
+#[cfg_attr(target_arch = "arm", target_feature(enable = "v7"))]
+#[cfg_attr(all(test, target_arch = "arm"), assert_instr(dup))]
+#[cfg_attr(all(test, target_arch = "aarch64"), assert_instr(dup))]
+pub unsafe fn vmovq_n_u8(value: u8) -> uint8x16_t {
+    vdupq_n_u8(value)
+}
+
+macro_rules! arm_reinterpret {
+    ($name:ident, $from:ty, $to:ty) => {
+        /// Vector reinterpret cast operation
+        #[inline]
+        #[target_feature(enable = "neon")]
+        #[cfg_attr(target_arch = "arm", target_feature(enable = "v7"))]
+        pub unsafe fn $name(a: $from) -> $to {
+            transmute(a)
+        }
+    };
+}
+
+// uint64x1_t vreinterpret_u64_u32 (uint32x2_t a)
+arm_reinterpret!(vreinterpret_u64_u32, uint32x2_t, uint64x1_t);
```

Comment:
It would be better not to use a macro here, otherwise stdarch-verify doesn't verify these.

Reply:
turned into functions.